src/Tokenomia/Vesting/PrivateSale.hs

```diff
@@ -0,0 +1,327 @@
+import           Data.ByteString.Lazy       (readFile)
+import           Data.ByteString.Internal   (unpackChars)
+import           Money                
+
+import qualified Blockfrost.Client      as B
+import qualified Blockfrost.Lens        as B
+
+import           Tokenomia.Common.Error
+import           Tokenomia.Common.Environment
+import           Tokenomia.Common.Blockfrost (projectFromEnv'')
+
+instance FromJSON PrivateSale
+instance FromJSON PrivateInvestor 
+instance FromJSON Investment 
+instance FromJSON Tranche 
+
+type Percentage = Integer 
+type Duration   = Integer 
+
+instance Eq Text where 
+  x == y = True
+
+instance Eq B.TxHash where
+  x == y = True
+
+data PrivateSale
+  = PrivateSale
+    { _psAddress         :: B.Address    -- Treasury address
+    , _psStart           :: POSIXTime
+    , _psTranches        :: [Tranche]
+    , _psAssetClass      :: AssetClass   -- Vesting token
+    , _psInvestors       :: [PrivateInvestor]
+    } deriving (Generic, Show)
+
+data Tranche 
+  = Tranche
+    { _tranchePercentage :: Percentage
+    , _trancheDuration   :: Duration 
+    } deriving (Generic, Show)
+
+data PrivateInvestor
+  = PrivateInvestor
+    { _piAddress         :: B.Address
+    , _piAllocation      :: Integer -- Amount of vesting tokens to lock (note, this is likely NOT equal to the total amounts of investments, as different tokens)
+    , _piInvestments     :: [Investment]
+    } deriving (Generic, Show)
+
+data Investment
+  = Investment
+    { _invTx             :: B.TxHash    -- Tx that sends below asset class to payment address
+    , _invAssetClass     :: AssetClass
+    , _invAmount         :: Integer -- Amount of above asset class expected to be sent to payment address
+    } deriving (Generic, Show)
+
+makeLenses ''PrivateSale
+makeLenses ''Tranche
+makeLenses ''PrivateInvestor
+makeLenses ''Investment
+
+verifyPrivateSale :: 
+  (MonadIO m
+  , MonadError  TokenomiaError m
+  , MonadReader Environment m)
+  => m ()
+verifyPrivateSale = do
+  liftIO . putStrLn $ "Please enter a filepath with JSON data"
+  jsonFilePath <- liftIO getLine
+  verifyPrivateSale' jsonFilePath
+
+verifyPrivateSale' :: 
+  (MonadIO m
+  , MonadError  TokenomiaError m
+  , MonadReader Environment m)
+  => FilePath 
+  -> m ()
+verifyPrivateSale' jsonFilePath = do
+  ps <- jsonToPrivateSale jsonFilePath
+  treasAddrTxs <- getTreasAddrTxs ps               -- :: [B.AddressTransaction] <- m [B.AddressTransaction] 
+  let invTxhs = privateSaleToTxhs ps in
+    let txhs = verifyTxHashList invTxhs treasAddrTxs in
+      if length txhs == length invTxhs then do
+        verifyTxs (getPsInvestments ps) txhs
+        else throwError . BlockFrostError . B.BlockfrostError $ "Unequal lengths, means missing TXs"
+
+jsonToPrivateSale ::
+  (MonadIO m
+  , MonadError  TokenomiaError m
+  , MonadReader Environment m)
+  => FilePath 
+  -> m PrivateSale
+jsonToPrivateSale jsonFilePath = do
+    fileContents <- liftIO . readFile $ jsonFilePath
```

Comment:
`Data.Aeon.decodeFileStrict` (actually `eitherDecodeFileStrict` is better since you'll get an actual error message about what went wrong during JSON parsing, then you can use `liftEither . Data.Bifunctor.first (BlockFrostError . B.BlockfrostError)`